File: python_backend/app/domain/entities/speaker.py
"""Speaker domain entity (B1).

A speaker registered for a session. Auto-created on first encounter of a
``deepgram_speaker_id``; user can rename via the rename endpoint."""
from dataclasses import dataclass
from typing import Optional


@dataclass
class Speaker:
    id: int
    session_id: str
    deepgram_speaker_id: int
    label: Optional[str]
    is_user: bool

File: web/src/domain/entities/speaker.ts
/**
 * Speaker domain entity — backend `SpeakerResponse` mirror (B3).
 *
 * Identity model:
 *   - `id` (positive integer) — backend PK. Assigned once the speaker is
 *     auto-registered by the WS pipeline.
 *   - `deepgramSpeakerId` — the raw 0/1/2... cluster id from Deepgram.
 *     Stable within a session; we key the live store by this value so
 *     we can render labels even before the backend has assigned a `id`.
 *
 * `colorHint` is one of the four Auri scenario colors (cyan / amber /
 * lavender / lime). The frontend resolves the semantic color name to
 * actual OKLCH via `design-system/tokens/colors.ts`.
 */

import type { ScenarioColor } from "./scenario";

export interface Speaker {
  id: number;
  sessionId: string;
  deepgramSpeakerId: number;
  label: string | null;
  isUser: boolean;
  colorHint: ScenarioColor;
}
